include/igl/principal_curvature.cpp

```diff
@@ -522,8 +522,12 @@ IGL_INLINE void CurvatureCalculator::getSphere(const int start, const double r,
       }
     }
   }
-  free(extra_candidates);
-  free(queue);
+  while(!extra_candidates->empty()){
```

Comment:
I believe this while and the clear are unnecessary. The `delete` operator should call the destructor of both variables. Honestly looking at the code you might as well remove the `new` operators altogether (and this is also true for the `bool* visited` which is `calloc`ed.

Reply:
[You're right](https://en.cppreference.com/w/cpp/container/priority_queue/~priority_queue), my C++ skills are a bit rusty. I'll commit the changes to the function. Unnecessary allocations with `new` happen in other code as well, which may need to be fixed in the future.

Reply:
BTW a potential fix for the slow AppVeyor build could be `/m:2` in the `MSBuildOptions` to use two threads for the two cores available in the VMs.